services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/DirectionsResponse.java

```diff
@@ -66,10 +66,16 @@ public static Builder builder() {
    * List of {@link DirectionsWaypoint} objects. Each {@code waypoint} is an input coordinate
    * snapped to the road and path network. The {@code waypoint} appear in the list in the order of
    * the input coordinates.
+   * @deprecated the waypoints are returned in the root of the response only if
+   *   {@link RouteOptions#waypointsPerRoute()} is false. Otherwise they are returned in
+   *   {@link DirectionsRoute#waypoints()}. You can continue to use this and it will work as-is.
+   *   However, it's recommended to use the new approach to distinguish waypoints between routes
+   *   (for example, this is necessary when asking for an EV-optimized route with alternatives).
    *
    * @return list of {@link DirectionsWaypoint} objects ordered from start of route till the end
    * @since 1.0.0
    */
+  @Deprecated
```

Comment:
The API will actually only be available in 2 weeks or so, so let's skip the deprecation notice for now.

Reply:
should i also remove a docs part?
```
 @deprecated the waypoints are returned in the root of the response only if
   *   {@link RouteOptions#waypointsPerRoute()} is false. Otherwise they are returned in
   *   {@link DirectionsRoute#waypoints()}. You can continue to use this and it will work as-is.
   *   However, it's recommended to use the new approach to distinguish waypoints between routes
   *   (for example, this is necessary when asking for an EV-optimized route with alternatives).
```

Reply:
Yes, we can re-add it later.